tests/ui/traits/const-traits/assoc-type-const-bound-usage-fail-2.stderr

```diff
@@ -1,10 +1,10 @@
-error[E0277]: the trait bound `<T as Trait>::Assoc<U>: ~const Trait` is not satisfied
+error[E0277]: the trait bound `U: ~const Other` is not satisfied
```

Comment:
Ideally these would be reported as derived causes, i.e. `U: ~const Other` must hold for `<T as Trait>::Assoc<U>: ~const Trait` to hold. I'm not going to touch the obligation cause code further in this PR though.